### PR TITLE
Kc/agent config v2 v3 docs

### PR DESCRIPTION
**Pull request by KristopherPaulsen**
I _could_ dry this up more, but I'm err'ing on the side of "duplicated but simple", as according to Niji, we're going to "change these up significantly soon" anyway.

### CODE REVIEW

File: docs/agent/config/_global-configuration.mdx [hunk 1]
```diff
@@ -0,0 +1,234 @@
+## Global Configuration
```
Comment (KristopherPaulsen) [inline docs/agent/config/_global-configuration.mdx:1]: Global pulled out into it's own shared section

File: docs/agent/config/index.mdx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+---
```
Comment (KristopherPaulsen) [inline docs/agent/config/index.mdx:1]: splash page as-per nijis request :salute

File: docs/agent/config/v2.mdx [hunk 3]
```diff
@@ -1,43 +1,11 @@
 ---
-title: Configuration File
+title: "Version 2"
```
Comment (KristopherPaulsen) [inline docs/agent/config/v2.mdx:2]: this is "mostly"™️ the same as the original page with some fiddly changes

- pulled out global
- fixed some copy-text to be less confusing now that we have v2/v3
- fixed links (now that we have v2/v3)
- etc

File: docs/agent/config/v3.mdx [hunk 4]
```diff
@@ -0,0 +1,571 @@
+---
+title: "Version 3"
+---
+
+import GlobalConfiguration from "./_global-configuration.mdx";
+
+## Full Example
```
Comment (KristopherPaulsen) [inline docs/agent/config/v3.mdx:7]: This "Full example" section is the main set of changes, where I:

- removed references to sections that no longer make sense under `endpoints`
- updated the examples to use the new `endpoints` formats, with relevant updated examples for each new case
- fixed links now that we have v2/v3
- fixed copy-text to mark some items as "deprecated", and with explanatory text.

File: docs/errors/_err_ngrok_list.md [hunk 5]
```diff
@@ -10207,7 +10207,7 @@
         <a id="ERR_NGROK_15002" href="/docs/errors/err_ngrok_15002">ERR_NGROK_15002</a>
       </td>
       <td class="pre-wrapped">
-        Your account is on the <code>&lt;PLAN&gt;</code> plan, which requires you to register a Domain or TCP Address through the ngrok Dashboard before starting an Endpoint. Visit <a href="https://dashboard.ngrok.com/cloud-edge/domains">https://dashboard.ngrok.com/cloud-edge/domains</a>, for HTTP and TLS endpoints, or <a href="https://dashboard.ngrok.com/cloud-edge/tcp-addresses">https://dashboard.ngrok.com/cloud-edge/tcp-addresses</a>, for TCP endpoints, to register a domain or address. Then use the `--url` flag when starting an HTTP or TLS tunnel with your Agent. For TCP tunnels, use `--remote_addr`.
```
Comment (KristopherPaulsen) [inline docs/errors/_err_ngrok_list.md:10210]: This section was missed in a previous set of changes, so included for consistency.

File: docs/agent/config/v3.mdx [hunk 6]
```diff
@@ -0,0 +1,571 @@
+---
+title: "Version 3"
+---
+
+import GlobalConfiguration from "./_global-configuration.mdx";
```
[the remark this answers is not in the document]
Comment (KristopherPaulsen) [reply docs/agent/config/v3.mdx:5]: You know, I don't think I actually made the connection they needed to be _under the agent key_. I can add a preamble, and maybe indent a bit to guide the user better here 👍

File: docs/agent/config/v3.mdx [hunk 7]
```diff
@@ -0,0 +1,584 @@
+---
+title: "Version 3"
```
Comment (russorat) [inline docs/agent/config/v3.mdx:2]:
```suggestion
title: "ngrok Agent Configuration File Version 3"
```

just adding some more context to the title.

File: docs/agent/config/v3.mdx [hunk 8]
```diff
@@ -0,0 +1,584 @@
...
+
+_For a complete list of fields, checkout [agent configuration](/docs/agent/config/v3/#agent-configuration)._
+
+## Full Example
+
+Below is an example configuration file with all the options filled in.
+To activate a specific option or endpoint, simply uncomment the relevant lines by removing the `#` at the beginning of the line.
+
+```yaml
+# ngrok Configuration File v3
```
Comment (russorat) [inline docs/agent/config/v3.mdx:27]:
```suggestion
# ngrok Agent Configuration File v3
```

File: docs/agent/config/v3.mdx [hunk 9]
```diff
@@ -0,0 +1,584 @@
...
+
+# ######################################################################
+# # Agent Configuration                                                #
+# ######################################################################
+
+# ----------------------------------------------------------------------
+# | Version  (required)                                                |
+# ----------------------------------------------------------------------
+
+# Specifies the version of the config file to use.
```
Comment (russorat) [inline docs/agent/config/v3.mdx:38]:
```suggestion
# Specifies the version of the config file to use. Valid options are 2 or 3.
```

File: docs/agent/config/v3.mdx [hunk 10]
```diff
@@ -0,0 +1,584 @@
...
+# ----------------------------------------------------------------------
+# | Connect Timeout                                                    |
+# ----------------------------------------------------------------------
+
+# How long to wait when establishing an agent session connection to the
+# ngrok service.
+#
+# Accepts duration, the default is 10s
+
+# connect_timeout: 30s
```
Comment (russorat) [inline docs/agent/config/v3.mdx:82]:
```suggestion
# connect_timeout: 10s
```

File: docs/agent/config/v3.mdx [hunk 11]
```diff
@@ -0,0 +1,584 @@
...
+# ----------------------------------------------------------------------
+# | Heartbeat Interval                                                 |
+# ----------------------------------------------------------------------
+
+# How often the ngrok agent should heartbeat to the ngrok servers defined
+# as a duration.
+#
+# Accepts a duration (e.g., 10s, 1m). The default value is `10s`.
+
+# heartbeat_interval: 1m
```
Comment (russorat) [inline docs/agent/config/v3.mdx:139]:
```suggestion
# heartbeat_interval: 10s
```

File: docs/agent/config/v3.mdx [hunk 12]
```diff
@@ -0,0 +1,584 @@
...
+# ----------------------------------------------------------------------
+# | Heartbeat Tolerance                                                |
+# ----------------------------------------------------------------------
+
+# This setting defines the maximum duration to wait for a heartbeat
+# response from the server before reconnecting the agent tunnel session.
+#
+# Accepts a duration (e.g., 10s, 1m). The default value is `15s`.
+
+# heartbeat_tolerance: 5s
```
Comment (russorat) [inline docs/agent/config/v3.mdx:150]:
```suggestion
# heartbeat_tolerance: 15s
```

File: docs/agent/config/v3.mdx [hunk 13]
```diff
@@ -0,0 +1,584 @@
...
+# ----------------------------------------------------------------------
+
+# This is the upper limit in bytes on memory to allocate when saving
+# requests over HTTP endpoints for inspection and reply.
+#
+# Accepts a numeric value. The default is `0`, equivalent to 50MB.
+#
+#   (!) To disable inspection for all endpoints, set the value to `-1`.
+
+# inspect_db_size: 104857600 # 100mb
```
Comment (russorat) [inline docs/agent/config/v3.mdx:163]:
```suggestion
# inspect_db_size: 52428800 # 50MB
```

File: docs/agent/config/v3.mdx [hunk 14]
```diff
@@ -0,0 +1,584 @@
...
+
+# ----------------------------------------------------------------------
+# | Log Level                                                          |
+# ----------------------------------------------------------------------
+
+# Sets the log detail level. Higher verbosity with each level.
```
Comment (russorat) [inline docs/agent/config/v3.mdx:169]: what's the default?

File: docs/agent/config/v3.mdx [hunk 15]
```diff
@@ -0,0 +1,584 @@
...
+#    info  - Informational messages.
+#   debug  - Detailed debugging info.
+
+# log_level: info
+
+# ----------------------------------------------------------------------
+# | Log Format                                                         |
+# ----------------------------------------------------------------------
+
+# Specifies the format of log records.
```
Comment (russorat) [inline docs/agent/config/v3.mdx:184]: what's the default?

File: docs/agent/config/v3.mdx [hunk 16]
```diff
@@ -0,0 +1,584 @@
...
+
+# ----------------------------------------------------------------------
+# | Log                                                                |
+# ----------------------------------------------------------------------
+
+# This is the destination where ngrok should write the logs.
+#
+# Allowed values:
+#   stdout  - write to standard out
+#   stderr  - write to standard error
```
Comment (russorat) [inline docs/agent/config/v3.mdx:201]: is this the default?
```suggestion
#   stderr  - write to standard error (default)
```

File: docs/agent/config/v3.mdx [hunk 17]
```diff
@@ -0,0 +1,584 @@
...
+
+# metadata: '{"serial": "00012xa-33rUtz9", "comment": "For customer alan@example.com"}'
+
+# ----------------------------------------------------------------------
+# | Proxy URL                                                          |
+# ----------------------------------------------------------------------
+
+# URL of an HTTP or SOCKS5 proxy for endpoint connections.
+#
+#   (!) ngrok will also respect the `http_proxy` and `http_proxy_env`
```
Comment (russorat) [inline docs/agent/config/v3.mdx:223]:
```suggestion
#   (!) ngrok will also respect the `http_proxy`, `http_proxy_env`, and `socks5_proxy`
```

File: docs/agent/config/v3.mdx [hunk 18]
```diff
@@ -0,0 +1,584 @@
...
+#
+# Allowed values:
+#    trusted - Use only the trusted certificate root for ngrok.com.
+#       host - Use the root certificates trusted by the host's OS (useful for MITM proxies with DPI).
+#     <path> - Path to a PEM file with additional trusted certificate authorities.
+
+# root_cas: trusted
+
+# ----------------------------------------------------------------------
+# | Server Address                                                     |
```
Comment (russorat) [inline docs/agent/config/v3.mdx:254]: i think agent.connect_address replaces this in v3

File: docs/agent/config/v3.mdx [hunk 19]
```diff
@@ -0,0 +1,584 @@
...
+| `proto`                                                                                  | `http`                                                                       | The endpoint protocol name. This defines the type of endpoint you would like to start.                                                                                                                                                                                                            |
+| `proxy_proto`                                                                            | String                                                                       | The version of [PROXY protocol](https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt) to use with this endpoint, empty if not using. Example values are 1 or 2.                                                                                                                            |
```
Comment (russorat) [inline docs/agent/config/v3.mdx:509]: this isn't in the "every option" example. might be ok, just calling that out.

File: docs/agent/config/v3.mdx (hunk 15 above)
Comment (KristopherPaulsen) [reply docs/agent/config/v3.mdx:184]: Good Q, I'll find out and add it

File: docs/agent/config/v3.mdx (hunk 14 above)
Comment (KristopherPaulsen) [reply docs/agent/config/v3.mdx:169]: Good Q, I'll find out and add it

File: docs/agent/config/v3.mdx (hunk 18 above)
Comment (KristopherPaulsen) [reply docs/agent/config/v3.mdx:254]: 😓  @warrn Do you happen to know off the top of your dome on this one?

File: docs/agent/config/v3.mdx [hunk 20]
```diff
@@ -0,0 +1,584 @@
...
+  #
+  # Start with after uncommenting:
+  #   ngrok start example-website
+
+  # - name: example-website
+  #   metadata: foo123
+  #   url: example-website.ngrok.app
+  #   upstream:
+  #     url: 8080
+  #     protocol: http1
```
[the remark this answers is not in the document]
Comment (KristopherPaulsen) [reply docs/agent/config/v3.mdx:366]: Good point. Thoughts? I could add an inline comment similar to what we mention at https://ngrok.com/docs/agent/cli/#flags-13.

Thoughts? 🤔

File: docs/agent/config/v3.mdx (hunk 19 above)
Comment (KristopherPaulsen) [reply docs/agent/config/v3.mdx:509]: Yeah, I think the "every option" example is still lagging behind a bit, good callout.

File: docs/agent/config/v3.mdx (hunk 18 above)
Comment (KristopherPaulsen) [reply docs/agent/config/v3.mdx:254]: Let me update this 👍

**Pull request merged**